client/asset/dcr/simnet_test.go

```diff
@@ -268,10 +270,12 @@ func runTest(t *testing.T, splitTx bool) {
 	}
 
 	confCoin := receipts[0].Coin()
+	confContract := receipts[0].Contract()
 	checkConfs := func(n uint32, expSpent bool) {
 		t.Helper()
-		confs, spent, err := rig.beta().Confirmations(tCtx, confCoin.ID())
-		if err != nil {
+		confs, err := rig.beta().SwapConfirmations(tCtx, confCoin.ID(), confContract, tStart)
+		spent := errors.Is(err, asset.ErrSpentSwap)
+		if err != nil || !spent {
```

Comment:
In this test I'm getting `simnet_test.go:292: error getting 0 confs: <nil>`.  Looks like the spent test is below where it compares to `expSpent`?

Reply:
Why not go back to a `spent bool` return from `SwapConfirmations` if it's less error prone and easier to reason about?

Reply:
I went back.